### PR TITLE
Connector commits before doing a potential merge

### PR DESCRIPTION
**Pull request by zachzhu2016**
connector framework should commit local changes before calling BriefcaseDb.pullAndMergeChanges() to prevent a potential merge from failing.

### CODE REVIEW

File: core/imodel-bridge/src/BridgeRunner.ts
Comment (calebmshafer) [inline core/imodel-bridge/src/BridgeRunner.ts:339]: @zachzhu2016 could you write a test that reproduces this issue?

Comment (zachzhu2016) [reply core/imodel-bridge/src/BridgeRunner.ts:341]: I don't think I can. It is an issue produced by parallel jobs. It is also a good practice to always commit changes (saveChanges()) before calling a function like pullAndMergeChanges() that could potentially modify the BriefcaseDb.

**Review by jffmarker — APPROVED** [review]

File: core/imodel-bridge/src/BridgeRunner.ts
Comment (calebmshafer) [reply core/imodel-bridge/src/BridgeRunner.ts:341]: There are strategies to simulate concurrency within tests and ensure the ordering and/or how often a method is called.  We use [sinon](https://sinonjs.org/) in a lot of places to assist us in things like this. We can chat about some strategies on how to do this when you have some time.

It'd be good for us to start building up a suite of tests around the iModel Bridge/Connector framework and this seems like the perfect example of where to start.

Comment (zachzhu2016) [reply core/imodel-bridge/src/BridgeRunner.ts:341]: Happy to chat more, but I'm not sure if it is appropriate to have that in this specific PR given it is always a better practice to save changes before merging, maybe in a separate PR. (even git doesn't let you merge if you had uncommitted local changes). I think that the only thing related to concurrency that the connector framework needs to protect is ensuring that all the operations done for individual change set are wrapped as an atomic operation and they are executed in the correct order (each individual operation's atomicity should already be guaranteed by IModelDb, though pushChanges() is not). It is as simple as calling several functions offered by IModelDb, mainly: 

1. lock channel
// let connector make change
2. acquire resources
3. save local changes
4. pull and merge changes
5. save potential merge changes
6. push changes

This PR adds step 2. But this is a separate discussion we can take offline.

Comment (calebmshafer) [reply core/imodel-bridge/src/BridgeRunner.ts:341]: @zachzhu2016 and I discussed some options here and with the potential changes coming to the framework we'll hold off on setting up a whole structure for testing. Depending what decision is made we'll re-visist.

**Comment by calebmshafer** [comment]
@Mergifyio update

**Pull request merged**
